internal/cmd/commands/scopescmd/list_keys.go

```diff
@@ -0,0 +1,150 @@
+package scopescmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/api/scopes"
+	"github.com/hashicorp/boundary/internal/cmd/base"
+	"github.com/mitchellh/cli"
+	"github.com/mitchellh/go-wordwrap"
+	"github.com/posener/complete"
+)
+
+var (
+	_ cli.Command             = (*ListKeysCommand)(nil)
+	_ cli.CommandAutocomplete = (*ListKeysCommand)(nil)
+)
+
+type ListKeysCommand struct {
+	*base.Command
+}
+
+func (c *ListKeysCommand) Synopsis() string {
+	return wordwrap.WrapString("List keys within a scope", base.TermWidth)
+}
+
+func (c *ListKeysCommand) Help() string {
+	return base.WrapForHelpText([]string{
+		"Usage: boundary scopes list-keys [args]",
+		"",
+		"  List keys within a scope. Example:",
+		"",
+		`    $ boundary scopes list-keys -scope-id global`,
+		"",
+		"",
+	}) + c.Flags().Help()
+}
+
+func (c *ListKeysCommand) Flags() *base.FlagSets {
+	set := c.FlagSet(base.FlagSetHTTP | base.FlagSetClient | base.FlagSetOutputFormat)
+	f := set.NewFlagSet("Command Options")
+
+	f.StringVar(&base.StringVar{
+		Name:   "scope-id",
+		Target: &c.FlagScopeId,
+		Usage:  "The id of the scope in which to list the keys",
+	})
+
+	return set
+}
+
+func (c *ListKeysCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictAnything
+}
+
+func (c *ListKeysCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *ListKeysCommand) printListTable(items []*scopes.Key) string {
+	if len(items) == 0 {
+		return "No keys found"
+	}
+	var output []string
+	output = []string{
+		"",
+		"Key information:",
+	}
+	for i, item := range items {
+		if i > 0 {
+			output = append(output, "")
+		}
+		if true {
+			output = append(output,
+				fmt.Sprintf("  ID:          %s", item.Id),
+				fmt.Sprintf("    Scope ID:  %s", item.Scope.Id),
+			)
+		}
+		if item.Version > 0 {
+			output = append(output,
+				fmt.Sprintf("    Version:   %d", item.Version),
+			)
+		}
+		if item.Type != "" {
+			output = append(output,
+				fmt.Sprintf("    Type:      %s", item.Type),
+			)
+		}
+		if item.Purpose != "" {
+			output = append(output,
+				fmt.Sprintf("    Purpose:   %s", item.Purpose),
+			)
+		}
+	}
+
+	return base.WrapForHelpText(output)
+}
```

Comment:
This looks like this:
```
$ boundary scopes list-keys -scope-id global

Key information:
  ID:          kdkv_DIYjdm5jm9
    Scope ID:  global
    Version:   1
    Type:      dek
    Purpose:   database

  ID:          kdkv_HOVS2nBVZO
    Scope ID:  global
    Version:   1
    Type:      dek
    Purpose:   oplog

  ID:          kdkv_LrbPzjZlLr
    Scope ID:  global
    Version:   1
    Type:      dek
    Purpose:   tokens

  ID:          kdkv_UP9ZFyf8iN
    Scope ID:  global
    Version:   1
    Type:      dek
    Purpose:   oidc

  ID:          kdkv_gZcLlzAMu6
    Scope ID:  global
    Version:   1
    Type:      dek
    Purpose:   sessions

  ID:          kdkv_r9QHRMtdUn
    Scope ID:  global
    Version:   1
    Type:      dek
    Purpose:   audit

  ID:          krkv_XDfWZanVZO
    Scope ID:  global
    Version:   1
    Type:      kek
    Purpose:   rootKey
```

Note that there are no authorized actions on keys, as actions are all done on the _scope_ not the individual key.